apps/cowswap-frontend/src/common/containers/OrderHooksDetails/index.tsx

```diff
@@ -0,0 +1,77 @@
+import { ReactElement, useMemo, useState } from 'react'
+
+import { latest } from '@cowprotocol/app-data'
+import { HookToDappMatch, matchHooksToDappsRegistry } from '@cowprotocol/hook-dapp-lib'
+
+import { ChevronDown, ChevronUp } from 'react-feather'
+
+import { AppDataInfo, decodeAppData } from 'modules/appData'
+
+import { HookItem } from './HookItem'
+import { HooksList, InfoWrapper, ToggleButton, Wrapper } from './styled'
+
+interface OrderHooksDetailsProps {
+  appData: string | AppDataInfo
+  children: (content: ReactElement) => ReactElement
+}
+
+export function OrderHooksDetails({ appData, children }: OrderHooksDetailsProps) {
+  const [isOpen, setOpen] = useState(false)
+  const appDataDoc = useMemo(() => {
+    return typeof appData === 'string' ? decodeAppData(appData) : appData.doc
+  }, [appData])
+
+  if (!appDataDoc) return null
+
+  const metadata = appDataDoc.metadata as latest.Metadata
+
+  const preHooksToDapp = matchHooksToDappsRegistry(metadata.hooks?.pre || [])
+  const postHooksToDapp = matchHooksToDappsRegistry(metadata.hooks?.post || [])
+
+  if (!preHooksToDapp.length && !postHooksToDapp.length) return null
```

Comment:
We can't really guarantee it'll be the latest.
How about older orders?
Maybe a type guard to make sure it contains the required fields before the casting?

Well, that's kind of taken care on lines 28, 29 and 31 🤔 
Just thinking whether this could cause a run time error down the line.

Reply:
>How about older orders?

Unfortunately, for them we will display this:

![image](https://github.com/user-attachments/assets/2ad2ef5f-f0cc-409d-92c2-0cd0c1edf30c)

Reply:
But we can check whether [dappid](https://github.com/cowprotocol/app-data/pull/66) field exists, if no - we just don't display info about hooks. Wdyt?

Reply:
Fixed it [here](https://github.com/cowprotocol/cowswap/pull/4938).

See `Skip hooks before the hooks store was introduced` comment